src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -158,6 +161,115 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
     return Status(StatusCode::INTERNAL_ERROR, "Unexpected error during Tensor creation");
 }
 
+static Status matFormatToImageFormat(const size_t& matFormat, mediapipe::ImageFormat::Format& imageFormat) {
+    switch (matFormat) {
+    case CV_8UC1:
+        imageFormat = mediapipe::ImageFormat::GRAY8;
+        break;
+    case CV_8UC3:
+        imageFormat = mediapipe::ImageFormat::SRGB;
+        break;
+    case CV_8UC4:
+        imageFormat = mediapipe::ImageFormat::SRGBA;
+        break;
+    case CV_16UC1:
+        imageFormat = mediapipe::ImageFormat::GRAY16;
+        break;
+    case CV_16UC3:
+        imageFormat = mediapipe::ImageFormat::SRGB48;
+        break;
+    case CV_16UC4:
+        imageFormat = mediapipe::ImageFormat::SRGBA64;
+        break;
+    case CV_8SC1:
+        imageFormat = mediapipe::ImageFormat::GRAY8;
+        break;
+    case CV_8SC3:
+        imageFormat = mediapipe::ImageFormat::SRGB;
+        break;
+    case CV_8SC4:
+        imageFormat = mediapipe::ImageFormat::SRGBA;
+        break;
+    case CV_16SC1:
+        imageFormat = mediapipe::ImageFormat::GRAY16;
+        break;
+    case CV_16SC3:
+        imageFormat = mediapipe::ImageFormat::SRGB48;
+        break;
+    case CV_16SC4:
+        imageFormat = mediapipe::ImageFormat::SRGBA64;
+        break;
+    case CV_32FC1:
+        imageFormat = mediapipe::ImageFormat::VEC32F1;
+        break;
+    case CV_32FC2:
+        imageFormat = mediapipe::ImageFormat::VEC32F2;
+        break;
+    // case CV_32FC4:
+    //     imageFormat = mediapipe::ImageFormat::VEC32F4;
+    //     break;
+    default:
+        return StatusCode::INTERNAL_ERROR;
+        break;
+    }
+    return StatusCode::OK;
+}
+
+static Status deserializeTensor(const std::string& requestedName, const KFSRequest& request, mediapipe::ImageFrame& outTensor) {
+    auto requestInputItr = request.inputs().begin();
+    auto status = getRequestInput(requestInputItr, requestedName, request);
+    if (!status.ok()) {
+        SPDLOG_ERROR("Getting Input failed");
+        return status;
+    }
+    if (request.raw_input_contents().size() == 0 || request.raw_input_contents().size() != request.inputs().size()) {
+        std::stringstream ss;
+        ss << "Cannot find data in raw_input_content for input with name: " << requestedName;
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("[servable name: {} version: {}] Invalid message structure - {}", request.model_name(), request.model_version(), details);
+        return Status(StatusCode::INVALID_MESSAGE_STRUCTURE, details);
+    }
+    auto inputIndex = requestInputItr - request.inputs().begin();
+    auto& bufferLocation = request.raw_input_contents().at(inputIndex);
+
+    if (requestInputItr->shape().size() != 3) {
+        SPDLOG_ERROR("Invalid Mediapipe Image input shape size. Expected: 3 Actual: {}", requestInputItr->shape().size());
```

Comment:
Do we validate that in MP path? If not we should log on debug level here. Add specific retcode.